### ISSUE TITLE
Unity exception with 2022.3.18 ambigouous reference while UnityInitializer script on android device. IOS and editor is working fine.

### ISSUE THREAD

**Issue by FarhanIshfaq** [aws/aws-sdk-net#3568]
### Describe the bug

AmbiguousMatchException: Ambiguous match found.
System.RuntimeType.GetMethodImplCommon (System.String name, System.Int32 genericParameterCount, System.Reflection.BindingFlags bindingAttr, System.Reflection.Binder binder, System.Reflection.CallingConventions callConv, System.Type[] types, System.Reflection.ParameterModifier[] modifiers) (at <00000000000000000000000000000000>:0)
System.RuntimeType.GetMethodImpl (System.String name, System.Reflection.BindingFlags bindingAttr, System.Reflection.Binder binder, System.Reflection.CallingConventions callConv, System.Type[] types, System.Reflection.ParameterModifier[] modifiers) (at <00000000000000000000000000000000>:0)
Amazon.Util.Internal.AndroidInterop.GetStaticJavaField[T] (System.String className, System.String methodName) (at <00000000000000000000000000000000>:0)
Amazon.Util.Internal.AmazonHookedPlatformInfo.Init () (at <00000000000000000000000000000000>:0)
Amazon.Util.Internal.AmazonHookedPlatformInfo.get_Instance () (at <00000000000000000000000000000000>:0)
Amazon.UnityInitializer.Awake () (at <00000000000000000000000000000000>:0)
UnityEngine.GameObject.AddComponent[T] () (at <00000000000000000000000000000000>:0)
Amazon.UnityInitializer.AttachToGameObject (UnityEngine.GameObject gameObject) (at <00000000000000000000000000000000>:0)
AWSManager.Initialize (System.String gToken) (at <00000000000000000000000000000000>:0)
Bootstrap.InitializeGoogleDependencies (Assets.SimpleGoogleSignIn.Scripts.UserInfo user, System.String googleToken) (at <00000000000000000000000000000000>:0)
Bootstrap+<callingInit>d__20.MoveNext () (at <00000000000000000000000000000000>:0)
UnityEngine.SetupCoroutine.InvokeMoveNext (System.Collections.IEnumerator enumerator, System.IntPtr returnValueAddress) (at <00000000000000000000000000000000>:0)
UnityEngine.GameObject:AddComponent()
Amazon.UnityInitializer:AttachToGameObject(GameObject)
AWSManager:Initialize(String)
Bootstrap:InitializeGoogleDependencies(UserInfo, String)
<callingInit>d__20:MoveNext()
UnityEngine.SetupCoroutine:InvokeMoveNext(IEnumerator, IntPtr)

I'm gettting this exception on android device while calling the  : Amazon.UnityInitializer.AttachToGameObject(gameObject);

### Regression Issue

- [ ] Select this option if this issue appears to be a regression.

### Expected Behavior

Should not throw exception.

### Current Behavior

mbiguousMatchException: Ambiguous match found.
System.RuntimeType.GetMethodImplCommon (System.String name, System.Int32 genericParameterCount, System.Reflection.BindingFlags bindingAttr, System.Reflection.Binder binder, System.Reflection.CallingConventions callConv, System.Type[] types, System.Reflection.ParameterModifier[] modifiers) (at <00000000000000000000000000000000>:0)
System.RuntimeType.GetMethodImpl (System.String name, System.Reflection.BindingFlags bindingAttr, System.Reflection.Binder binder, System.Reflection.CallingConventions callConv, System.Type[] types, System.Reflection.ParameterModifier[] modifiers) (at <00000000000000000000000000000000>:0)
Amazon.Util.Internal.AndroidInterop.GetStaticJavaField[T] (System.String className, System.String methodName) (at <00000000000000000000000000000000>:0)
Amazon.Util.Internal.AmazonHookedPlatformInfo.Init () (at <00000000000000000000000000000000>:0)
Amazon.Util.Internal.AmazonHookedPlatformInfo.get_Instance () (at <00000000000000000000000000000000>:0)
Amazon.UnityInitializer.Awake () (at <00000000000000000000000000000000>:0)
UnityEngine.GameObject.AddComponent[T] () (at <00000000000000000000000000000000>:0)
Amazon.UnityInitializer.AttachToGameObject (UnityEngine.GameObject gameObject) (at <00000000000000000000000000000000>:0)
AWSManager.Initialize (System.String gToken) (at <00000000000000000000000000000000>:0)
Bootstrap.InitializeGoogleDependencies (Assets.SimpleGoogleSignIn.Scripts.UserInfo user, System.String googleToken) (at <00000000000000000000000000000000>:0)
Bootstrap+<callingInit>d__20.MoveNext () (at <00000000000000000000000000000000>:0)
UnityEngine.SetupCoroutine.InvokeMoveNext (System.Collections.IEnumerator enumerator, System.IntPtr returnValueAddress) (at <00000000000000000000000000000000>:0)
UnityEngine.GameObject:AddComponent()
Amazon.UnityInitializer:AttachToGameObject(GameObject)
AWSManager:Initialize(String)
Bootstrap:InitializeGoogleDependencies(UserInfo, String)
<callingInit>d__20:MoveNext()
UnityEngine.SetupCoroutine:InvokeMoveNext(IEnumerator, IntPtr)

### Reproduction Steps

Just use : Amazon.UnityInitializer.AttachToGameObject(gameObject);

### Possible Solution

To Work Fine with init.

### Additional Information/Context

No

### AWS .NET SDK and/or Package version used

AWSSDK unity updated.

### Targeted .NET Platform

.Net 2.1

### Operating System and version

Android.

**Comment by ashishdhingra** [comment]
@FarhanIshfaq Good morning. Please refer [AmbiguousMatchException](https://learn.microsoft.com/en-us/dotnet/api/system.reflection.ambiguousmatchexception?view=net-9.0); the `AmbiguousMatchException` is thrown when a member is invoked late-bound and multiple overloads satisfy the binding criteria, or when more than one member matches the binding criteria passed to a reflection method that can return only a single result. Please check your project file if it's attempting to load multiple assemblies containing similar type definition for Android target.

Also refer [Special considerations for Unity support](https://docs.aws.amazon.com/sdk-for-net/v3/developer-guide/unity-special.html) for guidance on how to use AWSSDK assemblies in your Unity application.

I found an article on Unity forums [Why do I get Error: Ambiguous Match Found (Non-Mediation PSR in use)?](https://support.unity.com/hc/en-us/articles/4408586248340-Why-do-I-get-Error-Ambiguous-Match-Found-Non-Mediation-PSR-in-use) providing resolution for similar issue, if that helps.

Thanks,
Ashish

**Comment by FarhanIshfaq** [comment]
Hi
Hope you are doing well.

Thanks for your response and sending me the solution you have got I tried
with it if it works

Regards

**Comment by FarhanIshfaq** [comment]
Hi Ashish
how are you?


i already did check all guidance about unity and aws but not anything is
working if use .net standart 2.1 dll's then they just make auth with
cognito but i can't make dataset to amazon cognito to sync  user data with
amazon cognito because in these dll's there is no class reference for
saving the data on cognito Please share a guide if you can .  i also did
check the aws unity sdk which is 4 year's old.

Thanks. Regards

**Comment by ashishdhingra** [comment]
> they just make auth with
cognito but i can't make dataset to amazon cognito to sync  user data with
amazon cognito because in these dll's there is no class reference for
saving the data on cognito 

@FarhanIshfaq Good morning. Could you please elaborate your scenario? Looks like the original issue is resolved by using NET Standard 2.1 DLLs. what are you trying to sync with Cognito? And which class reference are you looking for? Have you tried using `AWSSDK.CognitoSync` DLL included in https://sdk-for-net.amazonwebservices.com/latest/v3/aws-sdk-netstandard2.0.zip?

Thanks,
Ashish

**Comment by FarhanIshfaq** [comment]
> > they just make auth with
> > cognito but i can't make dataset to amazon cognito to sync  user data with
> > amazon cognito because in these dll's there is no class reference for
> > saving the data on cognito
> 
> @FarhanIshfaq Good morning. Could you please elaborate your scenario? Looks like the original issue is resolved by using NET Standard 2.1 DLLs. what are you trying to sync with Cognito? And which class reference are you looking for? Have you tried using `AWSSDK.CognitoSync` DLL included in https://sdk-for-net.amazonwebservices.com/latest/v3/aws-sdk-netstandard2.0.zip?
> 
> Thanks, Ashish

Yes i did use these sdk too but i didn't find CognitoSync.SyncManger file reference can you please guide a little how i can sync data on amazon cognito using these sdk on. 

Thanks for the reponse.

**Comment by vuquangdai23** [comment]
I have the same issue when building with Unity 2022.3.62f2. If anyone has a solution, please help us

**Comment by normj** [comment]
Closing because the Cognito Sync Manager has been deprecated and removed from the SDK since version 3.5 of the AWS .NET SDK. https://docs.aws.amazon.com/sdk-for-net/v3/developer-guide/net-dg-v35.html. The Cognito service discourages users to use Cognito Sync and recommend trying AWS AppSync. https://docs.aws.amazon.com/cognito/latest/developerguide/cognito-sync.html
